ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -141,34 +145,22 @@ void *libX11_Variadic_u64(uint64_t a_0, size_t count, unsigned long *list, void
       # Store how much data we added to the stack in our callee saved register we stole
       mov x28, x10
 
-      # Subtract one member due to nullptr ender
-      sub x10, x1, 1
-
-      # x11 - stack offset
+      # x11 - stack offset - for stack Arguments
       mov x11, sp
 
-      # x12 - load offset
+      # x12 - load offset into input array (skipping the first 7 arguments)
       add x12, x9, (7 * 8)
 
-      cmp x10, 1
-      b.eq .single%=
-      b.lt .no_single%=
+      # Compute the number of elements excluding the terminating nullptr
+      subs x10, x1, 1
```

Comment:
Can this instruction be folded into the `cmp x10, 1` below by changing it to `cmp x1, 2` and using x10 for the `ldr`/`str` instead?

Reply:
Likely just introduced a bug now since we're changing logic.